C23_WMDE_Desktop_DE_08

Abban opened the pull request:
- Banner is based on CTRL of last test.
- VAR banner only asks to upgrade when donation amount is under €25
- Adds a new AnimatedText component so we will no longer have to
  add if statements to check for the content of the visitorsVsDonors
  sentence.
- Add new step controller and test for main donation form that
  checks a max amount
- Add new button and test to display correct label depending on amount
- Add new banner tests for amount functionality

Ticket: https://phabricator.wikimedia.org/T348619

gbirke commented:
I'm closing this without merging - it would be a pain to merge into the right place in the commit history. Since this is a frontpage test, there is no big loss.

Abban committed:
Archive C23_WMDE_Desktop_DE_07

Ticket: Ticket: https://phabricator.wikimedia.org/T348619

Abban committed:
Prepare campaign for C23_WMDE_Desktop_DE_08

- Banner is based on CTRL of last test.
- Adds a new AnimatedText component so we will no longer have to
  add if statements to check for the content of the visitorsVsDonors
  sentence.

Ticket: https://phabricator.wikimedia.org/T348619

Abban committed:
Implement VAR for C23_WMDE_Desktop_DE_08

VAR banner only asks to upgrade when donation amount is under €25

- Add new step controller and test for main donation form that
  checks a max amount
- Add new button and test to display correct label depending on amount
- Add new banner tests for amount functionality

Ticket: https://phabricator.wikimedia.org/T348619

moiikana commented on banners/desktop/styles/styles.scss:
```diff
@@ -8,6 +8,9 @@
 @use 'src/themes/Treedip/defaults';
 @use 'src/themes/Treedip/ButtonClose/ButtonClose';
+@use 'src/themes/Treedip/ProgressBar/ProgressBar' with (
+	$progress-bar-margin: 0 15px
```
what's the reason for this not going into the theme?

Abban replied:
The iPad used to use the same theme with a different margin IIRC. It can probably be defaulted to `0 15px` in the theme now.
